src/hacbs/components/Commits/CommitDetailsView.tsx

```diff
@@ -71,19 +92,51 @@ const CommitDetailsView: React.FC<CommitDetailsViewProps> = ({ commitName, appli
     [loaded, pipelineruns],
   );
 
-  const status = React.useMemo(
-    () =>
-      loaded &&
-      Array.isArray(pipelineruns) &&
-      pipelineruns.length > 0 &&
-      pipelineRunFilterReducer(
-        pipelineruns.sort(
-          (a, b) =>
-            parseInt(a.metadata.creationTimestamp, 10) - parseInt(b.metadata.creationTimestamp, 10),
-        )[pipelineruns.length - 1],
-      ),
-    [loaded, pipelineruns],
-  );
+  const sortedPLRList: SortedPLRList = React.useMemo(() => {
+    if (!loaded || loadErr) {
+      return null;
+    }
+
+    const runs = {
+      [runStatus.Running]: [],
+      [runStatus.Cancelled]: [],
+      [runStatus.Failed]: [],
+      [runStatus.Pending]: [],
+      [runStatus.Succeeded]: [],
+    };
+
+    pipelineruns.map((plr) => {
+      // sort plr into respective status array
+      const plrStatus = pipelineRunFilterReducer(plr);
+      // put task in correct status array
+      statuses.map((status) => {
+        if (plrStatus === status) {
+          runs[status].push(plr);
+        }
+      });
```

Comment:
No, the `statuses` array holds the order specified by UX for different `runStatuses`

Reply:
Sorry I corrected my suggestion. Not sure why i wrote `runStatus` while i meant `run`. 
See now.

The suggestion is to simply look at the current `run` status map and if it contains a value, we push to that array rather than having to perform a loop to find a match with `statuses` since `run` should have already be initialized to only contain entries that match those in `statuses`.